### PR TITLE
Add custom queries to cosmwasm

### PR DESCRIPTION
**Pull request by affanv14**
### Changelist
-Adds an function in go to query the perpetual clob details
- Adds code in contract to for custom queries
- Adds binding in go and rust

### Test Plan
- tested locally, will add more tests in subsequent prs

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/docker-compose.yml [hunk 1]
```diff
@@ -116,7 +116,7 @@ services:
     image: ghcr.io/skip-mev/slinky-sidecar:v0.4.1
     entrypoint: >
       sh -c "slinky-config --chain dydx --node-http-url http://dydxprotocold0:1317 --oracle-config-path /etc/oracle.json &&
-             slinky --oracle-config-path /etc/oracle.json --update-market-config-path /etc/market.json"
+             slinky --oracle-config-path /etc/oracle.json --update-market-config-path /etc/market.json --log-level error"
```
Comment (teddyding) [inline protocol/docker-compose.yml:119]: Did you mean to include this?

File: protocol/dydx-cosmwasm/src/querier.rs [hunk 2]
```diff
@@ -15,11 +17,41 @@ impl<'a> DydxQuerier<'a> {
...
         .into();
 
-        self.querier.query(&request)
+        let result: StdResult<MarketPrice> = self.querier.query(&request);
+        result.map(|market_price| MarketPriceResponse { market_price })
+    }
+
+    pub fn query_subaccount(&self, owner: String, number: u32) -> StdResult<SubaccountResponse> {
+        // print the owner and number
+        println!("owner: {}, number: {}", owner, number);
```
Comment (teddyding) [inline protocol/dydx-cosmwasm/src/querier.rs:31]: Remove/

File: protocol/x/clob/keeper/clob_pair.go [hunk 3]
```diff
@@ -693,3 +693,35 @@ func (k Keeper) IsPerpetualClobPairActive(
 
 	return clobPair.Status == types.ClobPair_STATUS_ACTIVE, nil
 }
+
+func (k Keeper) GetPerpetualClobDetails(
```
Comment (teddyding) [inline protocol/x/clob/keeper/clob_pair.go:697]: Nit: could you a TODO for unit test

**Review by teddyding — APPROVED** [review]

File: protocol/docker-compose.yml (hunk 1 above)
Comment (affanv14) [reply protocol/docker-compose.yml:119]: Yes.. this already exists on main branch(or a version of it) which can be resolved via a merge conflict resolution

File: protocol/dydx-cosmwasm/src/querier.rs (hunk 2 above)
Comment (affanv14) [reply protocol/dydx-cosmwasm/src/querier.rs:31]: Good catch. Thanks!

File: protocol/x/clob/keeper/clob_pair.go (hunk 3 above)
Comment (affanv14) [reply protocol/x/clob/keeper/clob_pair.go:697]: Will do

File: protocol/docker-compose.yml (hunk 1 above)
Comment (affanv14) [reply protocol/docker-compose.yml]: Actually, now that I can build without docker, I dont need this anymore, removing

**Pull request merged**